airflow-core/src/airflow/cli/commands/config_command.py

```diff
@@ -704,3 +705,144 @@ def lint_config(args) -> None:
+            airflow config update --dry-run
+
+    :param args: The CLI arguments for updating configuration.
+    """
+    console = AirflowConsole()
+    changes_applied = []
+    modifications = ConfigModifications()
+
+    update_sections = args.section if args.section else None
+    update_options = args.option if args.option else None
+    ignore_sections = args.ignore_section if args.ignore_section else []
+    ignore_options = args.ignore_option if args.ignore_option else []
+
+    config_dict = conf.as_dict(
+        display_source=True,
+        include_env=True,
+        include_cmds=True,
+        include_secret=True,
+    )
+    for change in CONFIGS_CHANGES:
+        conf_section = change.config.section.lower()
+        conf_option = change.config.option.lower()
+        full_key = f"{conf_section}.{conf_option}"
+
+        if update_sections is not None and conf_section not in [s.lower() for s in update_sections]:
+            continue
+        if update_options is not None and full_key not in [opt.lower() for opt in update_options]:
+            continue
+        if conf_section in [s.lower() for s in ignore_sections] or full_key in [
+            opt.lower() for opt in ignore_options
+        ]:
+            continue
+
+        if conf_section not in config_dict or conf_option not in config_dict[conf_section]:
+            continue
+        value_data = config_dict[conf_section][conf_option]
+        if not (isinstance(value_data, tuple) and value_data[1] == "airflow.cfg"):
+            continue
+
+        current_value = value_data[0]
+
+        if change.default_change:
+            if str(current_value) != str(change.new_default):
+                modifications.add_default_update(conf_section, conf_option, str(change.new_default))
+                changes_applied.append(
+                    f"Updated default value of '{conf_section}/{conf_option}' from '{current_value}' to '{change.new_default}'."
+                )
+        if change.renamed_to:
+            modifications.add_rename(
+                conf_section, conf_option, change.renamed_to.section, change.renamed_to.option
+            )
+            changes_applied.append(
+                f"Renamed '{conf_section}/{conf_option}' to '{change.renamed_to.section.lower()}/{change.renamed_to.option.lower()}'."
+            )
+        elif change.was_removed:
+            modifications.add_remove(conf_section, conf_option)
+            changes_applied.append(f"Removed '{conf_section}/{conf_option}' from configuration.")
+
+    backup_path = AIRFLOW_CONFIG + ".bak"
+    try:
+        shutil.copy2(AIRFLOW_CONFIG, backup_path)
+        console.print(f"Backup saved as '{backup_path}'.")
+    except Exception as e:
+        console.print(f"Failed to create backup: {e}")
+        raise AirflowConfigException("Backup creation failed. Aborting update_config operation.")
+
+    if getattr(args, "dry_run", False):
```

Comment:
would like to confirm whether we really need to use `getattr` here?

Reply:
changed it